### DISCUSSION TITLE
Logging NPE?

### DISCUSSION THREAD

**Discussion by woj-tek** [casid/jte#402] [General]
Is there a way for jte to log NPEs?

I was just trying to figure out why part of the template is not rendering and it turned out that one call was throwing NPE but there was no log anywhere which made troubleshooting it more difficult (when I wrapped suspect I got the stacktrace just fine)

**Comment by casid** [comment]
Hi @woj-tek!

That is wild, do you know where exactly the exception got lost?

jte should wrap the npe in a TemplateException and also include the template line location info where the problem occured.

**Reply by woj-tek** [reply]
Hmm... it was a bit ago so I don't remember exact details. The template is just:

```
@if(option.getValue() != null)
    <option value="${option.getValue()}"
            selected="${formField.isSelected(option)}">${option.getLabel()}</option>
@else
    <option value=""
            selected="${formField.isSelected(option)}">${option.getLabel()}</option>
@endif
```

And the problem happened when calling `boolean isSelected(Option option)` method. Here's the signature:

```
public boolean isSelected(Option option) {
	return getValues().contains(option.getValue());
}
```

From the looks of it (and quick test) - if there is any NPE thrown inside `isSelected()` then the NPE is not shown anywhere, eg:

```
public boolean isSelected(Option option) {
	String bla = null;
	System.out.println(bla);
	bla.contains("bla");
}
```

**Reply by casid** [reply]
How do you use jte? Is there maybe a web framework involved that swallows this exception?

When using jte directly, this will throw an exception. For instance, this test demonstrates that behavior: https://github.com/casid/jte/blob/0c7d4ad7a135d169b2db85062d827279e6d86aaf/jte/src/test/java/gg/jte/TemplateEngineTest.java#L1106
